21-Hello-World/05-Application-Structure/src/02-MTL/01-Foundations.md/03-Special-Output.md

```diff
@@ -214,7 +214,7 @@ class (Apply f) <= Applicative f where
 
 instance (Apply (OutputBox input)) <= Applicative (OutputBox input) where
```

Comment:
On reflection, this should really be...
```diff
- instance (Apply (OutputBox input)) <= Applicative (OutputBox input) where
+ instance (Applicative (OutputBox input)) <= Applicative (OutputBox input) where
```
because we're using `Box`'s `pure`, but we'll only have access to that if our constraint is increased from `Apply` to `Applicative`.

This bug is likely due to a copy-paste error on my part.

Reply:
- instance (Apply (OutputBox input)) <= Applicative (OutputBox input) where
+ instance (Applicative (OutputBox input)) <= Applicative (OutputBox input) where

In the diff above `Applicative (OutputBox input)` is duplicated on either side of `<=`. Not sure I follow.

Reply:
Does this help?
```diff
- instance (Apply       (OutputBox input)) <=
+ instance (Applicative (OutputBox input)) <=
```
In my original code, we have the `Apply` constraint, which doesn't grant us access to the function, `pure`. By changing it to `Applicative`, we can use `pure` in the instance's body.

Reply:
I really don't know how to interpret what you wrote about `instance (Applicative (OutputBox input)) <= Applicative (OutputBox input)`. 

If the snippet, I have fixed in the pull-request, is a pseudo-code snippet, which combines the type class definition and the instance implementation for didactic reasons, I don't understand how `constraints` and `classNameAndFundeps` in `constraints '<=' classNameAndFundeps` could be the same. 

If the snippet is not written in pseudo-code, the instance head doesn't expect to contain the constraints part, thus the snippet  should be just:
```
instance Applicative (OutputBox input) where
  pure :: forall a. a -> (OutputBox input) a
  pure value = OutputBox (\_ -> pure value)
```

Please check the final part of [Try PureScript link](https://try.purescript.org/?gist=aa4d6e138c96e4caf14caf27f82b3829) or [the gist](https://gist.github.com/MitinPavel/aa4d6e138c96e4caf14caf27f82b3829) (it is written on Purescript 0.13 since Try PureScript doesn't support 0.14 yet).

Reply:
Ah, perhaps it would be better to continue this discussion in an issue. This PR was submitted to fix one thing and it fixed it. I derailed this PR by talking about something else I noticed in the meantime.

In short, the above instance you provided works. However, I'm trying to build a foundation for understanding monad transformers.